### DISCUSSION TITLE
How about using GNUstep as the source of our desktop window and Dock?

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#356] [Q&A]
![image](https://user-images.githubusercontent.com/44593430/162475180-0f1cf143-1937-4dfb-8613-75890d1df120.png)
Although it's not like the window manager that most people like, it was the main interface of NeXT and also has a small Dock, maybe it can be used to replace Kwin, just one challenge, it's C language.
e.g. Étoilé OS [[Github](https://github.com/etoile/)]
![image](https://user-images.githubusercontent.com/44593430/162477571-3a8e08ed-a991-4bbb-ab22-e1ae9e74f5bd.png)

**Comment by probonopd** [comment]
GNUstep was investigated as part of the [LIVEstep](https://github.com/probonopd/LIVEstep) experiment before we decided on Qt for helloSystem.

**Reply by louies0623** [reply]
What made you switch to Kwin?

**Reply by probonopd** [reply]
It is the only Qt based window manager I could find, and Qt is our native toolkit.

**Reply by louies0623** [reply]
And what about lumina? Basically another QT based, ever considered or tested.

**Reply by probonopd** [reply]
Yes: https://github.com/helloSystem/hello/wiki/Components

**Reply by louies0623** [reply]
Do you want to try building an experimental version of lumina？

**Reply by probonopd** [reply]
Why? I had tested it before helloSystem was developed.

**Reply by louies0623** [reply]
I understand.

**Reply by larshenrikoern** [reply]
Lumina is not having its own windowmanager using fluxbox i think. So not a qt based one

**Reply by larshenrikoern** [reply]
Other C++ based windowmanagers to consider might be a fork of one from here https://www.libhunt.com/l/cpp/topic/window-manager . As far I know qt and c++ work well together.

**Comment by louies0623** [comment]
I accidentally discovered this project before, maybe it can be used as the WM of the hello classic theme.
[MLVWM](https://www.google.com/url?sa=t&source=web&rct=j&url=https://github.com/morgant/mlvwm&ved=2ahUKEwiK8Ib90JH7AhUM7WEKHcgWAWsQFnoECFwQAQ&usg=AOvVaw2nth1km9qbtCaU0nn_uj8O)
![image](https://user-images.githubusercontent.com/44593430/199680984-303e3f48-c799-4898-bd48-2970f6dc8ece.png)
